Activity review 2 july232026 (#16)
ml4sts-outreach/lm-unplugged: activity.md

my_demo
```

The document has seven tokensL green, blue, purple, blue, pink, blue, white. 
We will begin training by looking at the first two stickies to start: 

```{code-cell} python
:tags: [remove-input]
my_demo.train_step(1)
```

> The first sticky is green (context) and the second is blue (current token). Find the green bin and drop the blue ball into it. 

Reviewer: let's make this automatic instead of hard coded

```{code-cell} python
:tags: [remove-input]
my_demo.train_step(4)
```

Blue is back to being the context and the current token is pink. Pink goes into the blue bin with the purple ball from a previous step. 

Notice that the blue bin has balls of different colors. This means that the model has learned that multiple different colors can follow blue.

```{code-cell} python
:tags: [remove-input]
my_demo.train_step(5)
```

Pink is the context and blue is the current token. A blue ball is dropped into the pink bin.

```{code-cell} python
:tags: [remove-input]
my_demo.train_step(6)
```

Blue is the context and white is the current token. A white ball goes into the blue bin. The blue bin now has three different balls--purple, pink, and white--which shows three different transitions that followed blue in this document. 

A white token was drawn therefore training is complete. Each transition has been recorded as a ball to the corresponding bins. The collection of filled bins is the trained model. Each bin contains....

## Variations

### Context Two

> We have so far used a model with a context length of one (model looks at most recent color before deciding), but real models have longer context windows--meaning models can capture more specific patterns like "after blue -> then purple -> maybe green".

Reviewer: ```suggestion
We have so far used a model with a context length of one (model looks at only the one past word to sample the next word), but real models have longer context windows--meaning models can capture more specific patterns like "after blue -> then purple -> maybe green".
```
